packages/components/src/components/wallet-icon/wallet-icon.tsx

```diff
@@ -46,16 +48,19 @@ const sizes = {
     },
 } as const;
 
-const WalletIcon = ({ currency, icon, size = 'medium', type, has_bg }: TWalletIconProps) => {
+const WalletIcon = ({ currency, icon, size = 'medium', type, has_bg, is_demo, is_dark }: TWalletIconProps) => {
     if (!icon) {
         return null;
     }
 
+    const active_currency_for_bg = is_demo ? 'demo' : currency;
```

Comment:
@hamid-deriv Just a small suggestion, We can ignore it 🙇🏻 
We are adding this `is_demo` in many places now, The same as what we did for wallet icons and so on. As I suggested before these kinds of things can be handled in the **data transformation** layer and the UI doesn't need to know about the edge cases.

Luckily @thisyahlen-deriv already added the `useWalletList` hook which is where all this data comes from, So why not move these things there?

_packages/hooks/src/useWalletsList.ts_
```ts
/** The gradient class name for the wallet. */
gradient_color: `wallet_${wallet.is_virtual === 1 ? 'demo' : wallet.currency}_gradient`,
```
And then:
```ts
<WalletIcon color={wallet.gradient_color}  />
```
We can do the same for icons, You can have a look at the `useCurrencyConfig` hook and the icon property there.

The code above is just an example tho 😅

cc: @nijil-deriv What do you think? 🤔

Reply:
@farzin-deriv It's really interesting :) 1 week ago when I've created hook useWalletAccounts which did the same as Thisyahlen's hook and add some properties like `name`, `icon_name`, `icon_type` and etc needed for components, and asked you about this, you said we need to use Thisyahlen's hook))) In result I needed to create util function `convertWallets` to add these properties to Thisyahlen's hook result...

Reply:
@sergei-deriv Hey buddy, Yes you are right, Both hooks do the same kinda, We decided to go with [useWalletList](https://github.com/binary-com/deriv-app/blob/cc0a80edb0e120960a4c966a7a9f59b707d6b7df/packages/hooks/src/useWalletsList.ts) over [useWalletAccounts](https://github.com/binary-com/deriv-app/blob/14023cb912ccf947cccc24016585e404fcf26bb0/packages/hooks/src/useWalletAccounts.ts) since it's already doing the API call instead of using existing stores, So less refactoring for later, And also better TS and documentation and unit test and performance 🚀 

We shouldn't take ownership of code, We should all work as a team and improve each other's work, If something missing in `useWalletList` we can improve it instead of creating our own thing to separate our work from others  🙇🏻

Let us know if we should set up a call with the team to resolve the conflict and misunderstanding 🙇🏻

Reply:
@farzin-deriv Hey man. It's not about me and not about my code, it's just about case when I tried to explain we need this code in the future)) But it's fully my problem, I really bad explain my point of view and I have misunderstanding all the time

Reply:
Hey @farzin-deriv, yes this was something we (me and @hamid-deriv) were discussing too.I feel the same as well, that it's not best practise to let the component know business logic with props like `is_demo` etc. Moving it to hooks sound like an interesting approach. 👍